funcx_endpoint/funcx_endpoint/endpoint/utils/__init__.py

```diff
@@ -0,0 +1,29 @@
+import re as _re
+import typing as t
+
+_url_user_pass_pattern = r"://([^:]+):[^@]+@"
+_url_user_pass_re = _re.compile(_url_user_pass_pattern)
+_urlb_user_pass_re = _re.compile(_url_user_pass_pattern.encode())
+
+_T = t.TypeVar("_T", str, bytes)
+
+
+def _redact_url_creds(raw: _T, redact_user=True, repl="***", count=0) -> _T:
```

Comment:
I'm surprised you were able to capture all the scenarios in so few lines.  Like it.  As regex is always a bit dense (and sometimes error prone), how about a test to check a few of the possible inputs to `assert sensitve_info not in redacted_results` ?  Should be a simple unit test.

Reply:
Thanks for holding my feet to the fire there.  I was originally thinking that the testing of this was encapsulated as part of other tests.  Namely, "were they anonymized?!".  But you're correct that for _unit_ testing, I should test it directly.  Done.  See `tests/unit/test_utils.py`.